index.js

```diff
@@ -12,17 +11,28 @@ function ngrokIsAvailable() {
   }
 }
 
-server.listen(port, function () {
-  console.log('listening on port ' + port);
+if (config.isValid()) {
+  var port = config.get("port") || 8080;
+  server.listen(port, function () {
+    console.log('listening on port ' + port);
 
-  if (process.env.NODE_ENV === 'development' && ngrokIsAvailable()) {
-    require('ngrok').connect(port, function (err, url) {
-      if (err) {
-        console.error('ngrok error', err);
-        return;
-      }
+    if (process.env.NODE_ENV === 'development' && ngrokIsAvailable()) {
+      require('ngrok').connect(port, function (err, url) {
+        if (err) {
+          console.error('ngrok error', err);
+          return;
+        }
+
+        console.log('publicly accessible https url is: ' + url);
+      });
+    }
+  });
+}
+else {
```

Comment:
I'd prefer this be early-return style instead of if/else style, since there's nothing left to do if the config is invalid. In other words:

```
if (!config.isValid()) {
  // do stuff
  return;
}

// do stuff when config is valid
```

this helps keep the nesting to a minimum